use-cases/NVIDIA-GPU-passthrough-and-Kata.md

```diff
@@ -0,0 +1,296 @@
+
+NVIDIA GPU Pass-Through mode (one VM to one physical GPU)  is one flavor of graphics virtualization approach. This flavor allows direct assignment of an entire GPU to a single user, passing the native driver capabilities through the hypervisor without any limitations.
+
+NVIDIA vGPU mode enables multiple virtual machines (VMs) to have simultaneous, direct access to a single physical GPU.
+
+| Technology | Description | Behaviour | Detail |
+| --- | --- | --- | --- |
+| NVIDIA GPU Pass-Through mode | GPU passthrough | Physical GPU assigned to a single VM | Direct GPU assignment to VM without limitation |
+| NVIDIA vGPU mode | GPU sharing | Physical GPU shared by multiple VMs | Mediated passthrough |
+
+## Hardware Requirements
+NVIDIA GPUs Recommended for Virtualization:
+
+- NVIDIA Tesla (T4, M10, P6, V100...)
+- NVIDIA Quadro RTX 6000/8000
+
+About large BARs device (MMIO space above 4G): 
+
+large BARs devices:  Tesla P100, K40m ...
+```
+$ lspci -s 04:00.0 -vv | grep Region
+      Region 0: Memory at c6000000 (32-bit, non-prefetchable) [size=16M]
+      Region 1: Memory at 383800000000 (64-bit, prefetchable) [size=16G] #above 4G
+      Region 3: Memory at 383c00000000 (64-bit, prefetchable) [size=32M]
+```
+
+non-large BARs devices:  Tesla T4, P4 ...
+```
+$ lspci -s 04:00.0 -vv | grep Region
+      Region 0: Memory at c4000000 (32-bit, non-prefetchable) [size=16M]
+      Region 1: Memory at 38ffc0000000 (64-bit, prefetchable) [size=256M] #less than 4G
+      Region 3: Memory at 38fff0000000 (64-bit, prefetchable) [size=32M]
+```
+
+The following steps outline the workflow for using a Nvidia GPU with Kata.
+## Host BIOS Requirements
+
+For large BARs devices, MMIO mapping above 4G address space should be enabled in the PCI configuration of the BIOS.
+Some hardware vendors use different names in BIOS, such as:
+
+- Above 4G Decoding
+- Memory Mapped I/O above 4GB
+- Memory Hole for PCI MMIO
+
+
+
+## Host Kernel Requirements
+The following configurations need to be enabled on your host kernel:
+
+- CONFIG_VFIO_IOMMU_TYPE1
+- CONFIG_VFIO
+- CONFIG_VFIO_PCI
+- CONFIG_VFIO_MDEV
+- CONFIG_VFIO_MDEV_DEVICE
+
+Your host kernel needs to be booted with `intel_iommu=on` on the kernel command line.
+
+## Install and configure Kata Containers
+To use this feature, you need Kata version 1.11.0 or above ([related PR](https://github.com/kata-containers/runtime/pull/2461)).
+Follow the [Kata Containers setup instructions](https://github.com/kata-containers/documentation/blob/master/install/README.md) to install the latest version of Kata.
+
+For non-large BARs devices (for example, Nvidia Tesla T4):
+The following configuration in the Kata `configuration.toml` file as shown below can work:
+```
+machine_type = "pc"
+
+hotplug_vfio_on_root_bus = true
+```
+
+For large BARs devices (for example, Nvidia Tesla P100):
+The following configuration in the Kata `configuration.toml` file as shown below can work:
+
+hotplug with **shpchp**:
```

Comment:
I we can find an explanation of what `shpchp` is, could we add a hotlink.

Reply:
Added full name.